tests/ExampleApps/VolumeDemoApp/main.cpp

```diff
@@ -0,0 +1,173 @@
+// Include Radium base application and its simple Gui
+#include <Gui/BaseApplication.hpp>
+#include <Gui/RadiumWindow/SimpleWindowFactory.hpp>
+
+// include the Engine/entity/component interface
+#include <Core/Geometry/MeshPrimitives.hpp>
+#include <Engine/Scene/EntityManager.hpp>
+#include <Engine/Scene/GeometryComponent.hpp>
+#include <Engine/Scene/GeometrySystem.hpp>
+
+#include <QTimer>
+
+int main( int argc, char* argv[] ) {
+    //! [Creating the application]
+    Ra::Gui::BaseApplication app( argc, argv );
+    app.initialize( Ra::Gui::SimpleWindowFactory {} );
+    //! [Creating the application]
+
+#ifndef DEMO_FILE
+    //! [Creating the Volume]
+    auto density = new Ra::Core::Geometry::VolumeGrid();
+    int sx       = 100;
+    int sy       = 100;
+    int sz       = 75;
+
+    density->setSize( Ra::Core::Vector3i( sx, sy, sz ) );
+    auto& data = density->data();
+    // Radial density with value 1 at center
+    auto densityField = [sx, sy, sz]( int i, int j, int k, int r ) {
+        i -= sx / 2;
+        j -= sy / 2;
+        k -= sz / 2;
+        Scalar d = ( std::sqrt( i * i + j * j + k * k ) - r ) / r;
+        if ( d > 0 ) { return 0_ra; }
+        else {
+            return -d;
+        }
+    };
+    for ( int i = 0; i < sx; ++i ) {
+        for ( int j = 0; j < sy; ++j ) {
+            for ( int k = 0; k < sz; ++k ) {
+                data[i + sx * ( j + sy * k )] = densityField( i, j, k, 50 );
+            }
+        }
+    }
+    auto volume    = new Ra::Core::Asset::VolumeData( "DemoVolume" );
+    volume->volume = density;
+
+    // from measured media at https://github.com/mmp/pbrt-v3/blob/master/src/core/medium.cpp
```

Comment:
Should be either removed or added as option (with keyboard key to switch material for instance)

Reply:
If the symbol DEMO_FILE is not defined, the synthetic spherical volume is used. So, there is an option at build time to choose between loading a volume file or building a synthetic volume.

Reply:
> (with keyboard key to switch material for instance)
You're welcome to commit and push any suggestion.